src/Engine/Manager/VRManager.cpp

```diff
@@ -128,15 +128,34 @@ void VRManager::SetScale(float scale) {
     this->scale = scale;
 }
 
-bool VRManager::GetInput(vr::EVRButtonId buttonID) {
-    for (vr::TrackedDeviceIndex_t unDevice = 0; unDevice < vr::k_unMaxTrackedDeviceCount; unDevice++) {
-        vr::VRControllerState_t controllerState;
-        if (vrSystem->GetControllerState(unDevice, &controllerState, sizeof(controllerState))) {
-            pressedTrackedDevice[unDevice] = controllerState.ulButtonPressed == 0;
-            if (controllerState.ulButtonPressed & vr::ButtonMaskFromId(buttonID))
-                return true;
+bool VRManager::GetInput(vr::EVRButtonId buttonID, int ID) {
+    if (ID == 1) {
+        for (vr::TrackedDeviceIndex_t unDevice = 0; unDevice < vr::k_unMaxTrackedDeviceCount; unDevice++) {
+            vr::ETrackedControllerRole role = vrSystem->GetControllerRoleForTrackedDeviceIndex(unDevice);
+            if (role == vr::ETrackedControllerRole::TrackedControllerRole_LeftHand) {
+                vr::VRControllerState_t controllerState;
+                if (vrSystem->GetControllerState(unDevice, &controllerState, sizeof(controllerState))) {
+                    pressedTrackedDevice[unDevice] = controllerState.ulButtonPressed == 0;
+                    if (controllerState.ulButtonPressed & vr::ButtonMaskFromId(buttonID))
+                        return true;
+                }
+            }
         }
     }
+    else if (ID == 2) {
```

Comment:
else on same line as bracket

Reply:
Actually, this entire else block is mostly a duplicate of the one above. We could store the desired role enum depending on `ID` and then check against that variable instead of hardcoding values. This way we can have one block that handles both controllers.